index.js

```diff
@@ -1,92 +1,132 @@
-var sift = require('sift'),
-    clues = require('clues'),
+var clues = require('clues'),
+    { pathParser, astToCluesPath, astToString } = require('./ast'),
     d3Scale = require('d3-scale'),
-    Promise = clues.Promise;
-
-var reSplitter = /[\||Λ]/;
-
-// Polyfill for Object.setPrototypeOf
-Object.setPrototypeOf = Object.setPrototypeOf || function(obj, proto) {
-  obj.__proto__ = proto;
-  return obj; 
-};
+    Promise = clues.Promise,
+    generateEvaluateConditionFn = require('./conditional');
 
 function setPrototype(self) {
   return function(d) {
     return Object.setPrototypeOf(d,Object.getPrototypeOf(self));
   };        
 }
-
-// WARNING Sift exposes access to javascript through $where
-// Here we override $where with an error
-sift.useOperator('where',function() { throw '$WHERE_NOT_ALLOWED';});
-
+ 
 // Helper functions
-function toDots(d) { return d.replace(/ᐉ/g,'.'); }
 function noop() {}
+function inverseIdentity(d) { return !d; }
+function identity(d) { return d; }
+function servicifyResult(maybeFn) {
+  // when boxed in an "external", we can't return a function or it'll try to solve it.  So if we get an $external back, 
+  // we have to make sure clues doesn't do its thing and properly mark it as a service method
+  if (typeof maybeFn === 'function' && maybeFn.name !== '$service') {
+    Object.defineProperty(maybeFn, 'name', { value: '$service'});
+  }
+  return maybeFn;
+}
 
 // This is the main prototype 
 var Query = Object.create(Array.prototype);
 
-Query.scale = function(_domain) {
+function createExternal(cb) {
+  let counter = 0,
+      context = {};
+
+  return function $external(path) { 
+    let ast = pathParser(path);
+    let firstNode = ast.root;
+    let firstNodeKey = `_counter${astToString(firstNode)}`;
+    let remainder = ast.extra ? `.${ast.extra}` : '';
+
+    let existingCounter = context[firstNodeKey];
```

Comment:
Might add a comment briefly explaining why we move the formula to `externalKey${existingCounter}`